lib/ws_servers/api/handlers/on_selected_algo_resume_remove.js

```diff
@@ -0,0 +1,55 @@
+const send = require('../../../util/ws/send')
+const validateParams = require('../../../util/ws/validate_params')
+const isAuthorized = require('../../../util/ws/is_authorized')
+const sendError = require('../../../util/ws/send_error')
+
+module.exports = async (server, session, msg) => {
+  const { d, algoDB } = server
+  const { AlgoOrder } = algoDB
+  const [, authToken, payload] = msg
+
+  const validRequest = validateParams(session, {
+    authToken: { type: 'string', v: authToken },
+    payload: { type: 'object', v: payload }
+  })
+
+  if (!validRequest) {
+    d('invalid request: selected_algo:resume_remove')
+    return
+  }
+
+  if (!isAuthorized(session, authToken)) {
+    return sendError(session, 'Unauthorized', ['unauthorized'])
+  }
+
+  const tasks = []
+
+  for (const [mode, { algoWorker }] of Object.entries(session.services)) {
+    if (algoWorker) {
+      tasks.push(...payload[mode].remove.map(({ gid }) => algoWorker.cancelOrder(gid)))
+
+      for (const { algoID, gid } of payload[mode].resume) {
+        const ao = await AlgoOrder.get({ algoID, gid })
+        const { state, createdAt } = ao || {}
+
+        if (!state) {
+          continue
+        }
+
+        tasks.push(algoWorker.host.loadAO(algoID, gid, JSON.parse(state), createdAt))
```

Comment:
I think you need to check first if the AO is not running already, reading from host and not from DB.